lib/rexml/parsers/baseparser.rb

```diff
@@ -349,8 +349,9 @@ def pull_event
         end
         begin
           @source.read if @source.buffer.size<2
-          if @source.buffer[0] == ?<
-            if @source.buffer[1] == ?/
+          pattern = @source.buffer[0,2]
```

Comment:
I think that this is not a pattern. Can we use more suitable name such as `buffer` and `next_data`?

Can we remove `[0,2]`? I think that it's a needless method call.